deps/rabbit/src/rabbit_amqqueue.erl

```diff
@@ -652,13 +652,47 @@ priv_absent(QueueName, QPid, _IsDurable, alive) ->
             'ok' | rabbit_types:channel_exit() | rabbit_types:connection_exit().
 
 assert_equivalence(Q, DurableDeclare, AutoDeleteDeclare, Args1, Owner) ->
-    QName = amqqueue:get_name(Q),
-    DurableQ = amqqueue:is_durable(Q),
-    AutoDeleteQ = amqqueue:is_auto_delete(Q),
-    ok = check_exclusive_access(Q, Owner, strict),
-    ok = rabbit_misc:assert_field_equivalence(DurableQ, DurableDeclare, QName, durable),
-    ok = rabbit_misc:assert_field_equivalence(AutoDeleteQ, AutoDeleteDeclare, QName, auto_delete),
-    ok = assert_args_equivalence(Q, Args1).
+    case ignore_checks_on_qq_redeclaration(Args1, Q) of
```

Comment:
I'd introduce a new function, `maybe_assert_equivalence/2`, and use it with  `rabbit_channel:with/2` (in `handle_method(#'queue.declare'…)`.

Reply:
Although, we do assert equivalence for both, just a slimmer version of it. I'll start with just renaming the check for 'which' equivalence we check and leave this function name as assert_equivalence for now.